src/AspNetCoreModuleV2/AspNetCore/proxymodule.cpp

```diff
@@ -112,15 +116,15 @@ ASPNET_CORE_PROXY_MODULE::OnExecuteRequestHandler(
             FINISHED_IF_FAILED(moduleContainer->SetConnectionModuleContext(static_cast<IHttpConnectionStoredContext*>(disconnectHandler.release()), m_moduleId));
         }
 
-        m_pDisconnectHandler->SetHandler(this);
-
         FINISHED_IF_FAILED(m_pApplicationManager->GetOrCreateApplicationInfo(
             *pHttpContext,
             m_pApplicationInfo));
 
         FINISHED_IF_FAILED(m_pApplicationInfo->CreateHandler(*pHttpContext, m_pHandler));
 
         retVal = m_pHandler->OnExecuteRequestHandler();
```

Comment:
OnExecuteRequestHandler may finished request synchronously. It will be too late to register the disconnect. This is why in the past, registering  disconnect notification is inside the handler.

Reply:
If it finishes sync we'll unregistered callback in the same method.